### PR TITLE
LEAF-4421 Simplify formGrid and example

### PR DESCRIPTION
**Pull request by mgaoVA**
This simplifies formGrid.js usage and updates the example.

Before:
```js
    formGrid.setData(Object.keys(queryResult).map(key => queryResult[key]));
    formGrid.setDataBlob(queryResult);
```

After:
```js
    formGrid.setDataBlob(queryResult);
```

To maintain backward compatibility: If setData() was not previously called, setDataBlob() will apply the same transformation from before.

## Potential Impact
Dependencies of formGrid.js

## Testing
- [ ] Homepage and Report builder works normally
- [ ] Custom implementations of formGrid.js such as LEAF Data Dictionary works correctly

### CODE REVIEW

**Review by jampaul3 — APPROVED** [review]

**Review by shaneodd — APPROVED** [review]

File: LEAF_Request_Portal/js/formGrid.js [hunk 1]
```diff
@@ -839,22 +840,40 @@ var LeafFormGrid = function (containerID, options) {
...
    * @memberOf LeafFormGrid
    */
   function setDataBlob(data) {
     dataBlob = data;
+    if(!usingSetData) {
+      if(Array.isArray(data)) {
+        currentData = data;
+      }
+      else {
+        currentData = Object.keys(data).map(key => data[key]);
```
Comment (Pelentan) [inline LEAF_Request_Portal/js/formGrid.js:870]: What happens when "data" is a string or null?

File: app/libs/js/LEAF/formGrid.js [hunk 2]
```diff
@@ -861,20 +840,40 @@ var LeafFormGrid = function (containerID, options) {
...
    * @memberOf LeafFormGrid
    */
   function setDataBlob(data) {
     dataBlob = data;
+    if(!usingSetData) {
+      if(Array.isArray(data)) {
+        currentData = data;
+      }
+      else {
+        currentData = Object.keys(data).map(key => data[key]);
```
Comment (Pelentan) [inline app/libs/js/LEAF/formGrid.js:870]: This function looks identical the the one I commented on above.  Do they need to be pulled out an put in a different file for both to reference one source?

File: libs/js/LEAF/formGrid.js [hunk 3]
```diff
@@ -861,20 +840,40 @@ var LeafFormGrid = function (containerID, options) {
...
    * @memberOf LeafFormGrid
    */
   function setDataBlob(data) {
     dataBlob = data;
+    if(!usingSetData) {
+      if(Array.isArray(data)) {
+        currentData = data;
+      }
+      else {
+        currentData = Object.keys(data).map(key => data[key]);
```
Comment (Pelentan) [inline libs/js/LEAF/formGrid.js:870]: Here it is again.  Normally the rule is if you have the same function that gets used 3 or more times, you want to pull it out to a central location, unless there is a good reason not to.

**Review by Pelentan — CHANGES_REQUESTED** [review]

**Review by aerinkayne — APPROVED** [review]

File: LEAF_Request_Portal/js/formGrid.js (hunk 1 above)
Comment (mgaoVA) [reply LEAF_Request_Portal/js/formGrid.js:870]: Updated to add a console error

File: app/libs/js/LEAF/formGrid.js (hunk 2 above)
Comment (mgaoVA) [reply app/libs/js/LEAF/formGrid.js:870]: The directory for this file is part of an incomplete migration

File: libs/js/LEAF/formGrid.js (hunk 3 above)
Comment (mgaoVA) [reply libs/js/LEAF/formGrid.js:870]: The directory for this file is part of an incomplete migration. Ideally it'd be a symlink.

**Review by Pelentan — APPROVED** [review]

**Pull request merged**